### PR TITLE
CI: run K8sServices on KIND

### PR DESCRIPTION
**Pull request by brb**
This PR allows us to run the `K8sServices` suite (probably others as well; I haven't tested it) on KIND on a fairly recent kernel (with the Daniel's cgroup fix http://github.com/torvalds/linux/commit/8520e224f547cd070c7c8f97b1fc6d58cff7ccaa) or the one running with cgroupv1 disabled.

HOWTO:
```
./contrib/scripts/kind.sh "" 2 "" "" none
export DOCKER_REGISTRY=localhost:5000
make dev-docker-image
docker tag localhost:5000/cilium/cilium-dev:latest localhost:5000/cilium/cilium-dev:local
docker push localhost:5000/cilium/cilium-dev:local
cd test
FOCUS="K8sServices" NETNEXT=true KUBEPROXY=0 K8S_NODES=3 NO_CILIUM_ON_NODES=kind-worker2 make k8s-kind
```

Please see commit msgs for more details.

### CODE REVIEW

File: contrib/scripts/kind.sh [hunk 1]
```diff
@@ -30,12 +32,12 @@ if ! have_kind; then
     echo "  https://kind.sigs.k8s.io/docs/user/quick-start/#installation"
 fi
 
-if [[ "${#}" -gt 4 ]]; then
+if [[ "${#}" -gt 5 ]]; then
```
Comment (nebril) [inline contrib/scripts/kind.sh:35]: nit: might as well move `5` to a variable

**Review by nebril — APPROVED** [review]

**Review by nbusseneau — APPROVED** [review]
Yes. :shipit:

File: contrib/scripts/kind.sh (hunk 1 above)
Comment (brb) [reply contrib/scripts/kind.sh:35]: Sure, will do it in a follow-up.

**Review by tklauser — APPROVED** [review]
:rocket:

**Review by joestringer — APPROVED** [review]
> ```
> export DOCKER_REGISTRY=localhost:5000
> make dev-docker-image
> docker tag localhost:5000/cilium/cilium-dev:latest localhost:5000/cilium/cilium-dev:local
> docker push localhost:5000/cilium/cilium-dev:local
> ```

I think the above is the same as `make kind-image`? (This also pre-loads the images into kind)

**Review by christarazi — APPROVED** [review]
Amazing :rocket:

**Pull request merged**